test/dom-parsing/XMLSerializer.tests.ts

```diff
@@ -0,0 +1,346 @@
+import * as chai from 'chai';
+import * as slimdom from '../../src/index';
+
+import { appendAttribute } from '../../src/util/attrMutations';
+
+const HTML_NAMESPACE = 'http://www.w3.org/1999/xhtml';
+const XML_NAMESPACE = 'http://www.w3.org/XML/1998/namespace';
+const XMLNS_NAMESPACE = 'http://www.w3.org/2000/xmlns/';
+
+describe('XMLSerializer', () => {
+	let document: slimdom.Document;
+	let serializer: slimdom.XMLSerializer;
+	beforeEach(() => {
+		document = new slimdom.Document();
+		serializer = new slimdom.XMLSerializer();
+	});
+
+	it('returns the empty string if given an Attr', () => {
+		chai.assert.equal(serializer.serializeToString(document.createAttribute('test')), '');
+		chai.assert.equal(
+			serializer.serializeToString(document.createAttributeNS('http://www.example.com/ns', 'test')),
+			''
+		);
+	});
+
+	it('can serialize a CDATASection', () => {
+		chai.assert.equal(serializer.serializeToString(document.createCDATASection('test')), '<![CDATA[test]]>');
```

Comment:
Done. It is unfortunately possible (through manipulating data after creation) to create a CDataSection containing `]]>`, which will result in invalid XML when serialized. The closest mention of this I could find is a work-around proposed in https://www.w3.org/Bugs/Public/show_bug.cgi?id=16682. The current behavior matches Chrome, Firefox, Edge and Safari, so I guess we should keep it for now and be careful when we mutate CDataSection data.